src/category_theory/sites/sieves.lean

```diff
@@ -355,8 +355,24 @@ lemma nat_trans_of_le_comm {S T : sieve X} (h : S ≤ T) :
 rfl
 
 /-- The presheaf induced by a sieve is a subobject of the yoneda embedding. -/
-instance functor_inclusion_is_mono : mono (functor_inclusion S) :=
+instance functor_inclusion_is_mono : mono S.functor_inclusion :=
 ⟨λ Z f g h, by { ext Y y, apply congr_fun (nat_trans.congr_app h Y) y }⟩
 
+/-- A natural transformation to a representable functor induces a sieve. -/
+-- TODO: show that when `f` is mono, this is inverse to `functor_inclusion`, in the appropriate
+-- sense.
+def sieve_of_subfunctor (R) (f : R ⟶ yoneda.obj X) : sieve X :=
```

Comment:
Should we have some simp-lemmas about this? Maybe just add `@[simps]` to the `def`?

Reply:
I thought about adding `simps`, but the simp-normal-form for `sieve.arrows` is with a coercion rather than the projection, so I think `simps` won't make it, and I haven't figured out how the new simps tricks can make this happen though this seems like the best solution. If not I don't mind just manually adding simp lemmas

Reply:
@fpvandoorn Could you please teach use the new `simps` trick?

Reply:
For clarity I've put the simp lemma that I think should be generated in `sieve_of_subfunctor_apply`

Reply:
Hmm, maybe it's not really in scope for `simps` to do things with `iff`-lemmas instead of just `rfl` "apply"-lemmas. So, let's consider the `simps`-thing "maybe later".

Reply:
Ah good point - fair enough! I guess the `=` version instead of the iff version would be okay too now that I think about it, but I'm happy either way